### PR TITLE
topsql: remove do topN before report

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #xxx

1. topsql: remove do topN before report
2. change default value of tidb_top_sql_max_statement_count from 200 to 100 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by crazycs520** [comment]
@breeswish @zhongzc PTAL

**Review by breeswish — APPROVED** [review]

File: util/topsql/reporter/datamodel.go [hunk 1]
```diff
@@ -496,29 +496,17 @@ func (c *collecting) appendOthersStmtStatsItem(timestamp uint64, item stmtstats.
...
 	for _, v := range c.records {
 		rs = append(rs, *v)
 	}
-	// Fetch TopN records.
-	var evicted records
-	rs, evicted = rs.topN(n)
-	if others != nil {
+	if others != nil && others.totalCPUTimeMs > 0 {
 		// Sort the records by timestamp to fix the affect of time jump backward.
 		sort.Sort(others)
```
Comment (zhongzc) [inline util/topsql/reporter/datamodel.go:509]:
```suggestion
```

**Comment by zhongzc** [comment]
rest LGTM

**Review by tangenta — APPROVED** [review]

**Review by breeswish — APPROVED** [review]

**Pull request merged**